subiquity/models/filesystem.py

```diff
@@ -689,10 +689,16 @@ def serialize_number(self):
 
     @property
     def _number(self):
-        if self.preserve:
```

Comment:
This seems a strange place to do this, so on the first access to a partitions _number attribute, all the partitions of the disk get assigned? Why not do it as the partitions are created? If there is a reason for this it should go into a comment :)